feat(dataarts): add dataarts architecture reviewer resource (#3902)
huaweicloud/terraform-provider-huaweicloud: docs/resources/dataarts_architecture_reviewer.md

---
subcategory: "DataArts Studio"
---

# huaweicloud_dataarts_studio_architecture_reviewer

Manages DataArts Studio architecture reviewer resource within HuaweiCloud.

## Example Usage

```hcl
variable "workspace_id" {}
variable "user_name" {}
variable "user_id" {}
variable "email" {}
variable "phone_number" {}

resource "huaweicloud_dataarts_studio_architecture_reviewer" "test" {
  workspace_id = var.workspace_id
  user_name    = var.user_name
  user_id      = var.user_id
  email        = "var.email"
  phone_number = "var.phone_number"
}
```

## Argument Reference

The following arguments are supported:

* `region` - (Optional, String, ForceNew) Specifies the region in which to create the architecture reviewer resource.
  If omitted, the provider-level region will be used. Changing this parameter will create a new resource.

> * `workspace_id` - (Required, String, ForceNew) Specifies the workspace ID in which the architecture reviewer.

Reviewer: ```suggestion
* `workspace_id` - (Required, String, ForceNew) Specifies the workspace ID to which the architecture reviewer belongs.
```

Author: down

  Changing this creates a new architecture reviewer.

* `user_name` - (Required, String, ForceNew) Specifies the user name of the architecture reviewer.

* `user_id` - (Required, String, ForceNew) Specifies the user ID of the architecture reviewer.

* `email` - (Optional, String, ForceNew) Specifies the email of the architecture reviewer.

* `phone_number` - (Optional, String, ForceNew) Specifies the phone number of the architecture reviewer.

## Attribute Reference

In addition to all arguments above, the following attributes are exported:

> * `id` - The resource ID, also the ID of the architecture reviewer.

Reviewer 2: ```suggestion
* `id` - The resource ID, the value is the `user_name`.
```

Author: down

> * `reviewer_id` - The ID of the reviewer to which the architecture reviewer belongs.

Reviewer 2: ```suggestion
* `reviewer_id` - The ID of the reviewer.
```

Author: down

## Import

> The DataArts architecture reviewer can be imported using the `workspace_id` and `id` separated by a slash, e.g.
> 
> ```bash
> $ terraform import huaweicloud_dataarts_architecture_reviewer.test <workspace_id>/<id>
> ```

Reviewer 2: ```suggestion
The DataArts architecture reviewer can be imported using the `workspace_id` and `user_name` separated by a slash, e.g.

```bash
$ terraform import huaweicloud_dataarts_architecture_reviewer.test <workspace_id>/<user_name>
```
```

Author: down

---
subcategory: "DataArts Studio"
---

# huaweicloud_dataarts_studio_architecture_reviewer

Manages DataArts Studio architecture reviewer resource within HuaweiCloud.

## Example Usage

```hcl

> variable "workspace_id" {}
> variable "user_name" {}
> variable "user_id" {}
> 
> resource "huaweicloud_dataarts_studio_architecture_reviewer" "test" {
>   workspace_id = var.workspace_id
>   user_name    = var.user_name
>   user_id      = var.user_id
> }
> 
> resource "huaweicloud_dataarts_studio_architecture_reviewer" "sub_test" {
>   workspace_id = var.workspace_id
>   user_name    = var.user_name
>   user_id      = var.user_id
>   email        = "var.email"
>   phone_number = "var.phone_number"
> }

Reviewer 2: ```suggestion
variable "workspace_id" {}
variable "user_name" {}
variable "user_id" {}
variable "email" {}
variable "phone_number" {}

resource "huaweicloud_dataarts_studio_architecture_reviewer" "test" {
  workspace_id = var.workspace_id
  user_name    = var.user_name
  user_id      = var.user_id
  email        = var.email
  phone_number = var.phone_number
}
```

Author: down

```

## Argument Reference

The following arguments are supported:

* `region` - (Optional, String, ForceNew) Specifies the region in which to create the architecture reviewer resource.
  If omitted, the provider-level region will be used. Changing this parameter will create a new resource.

* `workspace_id` - (Required, String, ForceNew) Specifies the workspace ID in which the architecture reviewer.
  Changing this creates a new architecture reviewer.

* `user_name` - (Required, String, ForceNew) Specifies the user name of the architecture reviewer.

* `user_id` - (Required, String, ForceNew) Specifies the user ID of the architecture reviewer.

* `email` - (Optional, String, ForceNew) Specifies the email of the architecture reviewer.

* `phone_number` - (Optional, String, ForceNew) Specifies the phone number of the architecture reviewer.

> * `email_notify` - (Optional, Bool, ForceNew) Specifies the email notify of the architecture reviewer.
> 
> * `sms_notify` - (Optional, Bool, ForceNew) Specifies the sms notify of the architecture reviewer.

Reviewer 2: ```suggestion
```

Author: down
